### PR TITLE
Automatic Update of models being served

### PR DESCRIPTION
**Pull request by deepak-matroid**
inception_inference_dynamic creates a service which dynamically scans directory to look for updated models. 
inception_inference.proto changed to allow for passing of network_id to run inference for a single network. This needs to be integrated from client side (matroid repo) to serve the right path and call classify with network_id

To test from terminal,
cd ~/serving
git pull origin
git checkout add_dynamic_inception_serving
git submodule update

##Serving Script
#Build
cd ~/serving/
bazel build  -c opt  //tensorflow_serving/inception_example:inception_inference_dynamic
#Run
~/serving/bazel-bin/tensorflow_serving/inception_example/inception_inference_dynamic --port=9031 ~/matroid/scheduler/model_serving/<model-id>/

##Client Script
#Build
cd ~/serving/
bazel build  -c opt  //tensorflow_serving/inception_example:inception_client
#Run
~/serving/bazel-bin/tensorflow_serving/inception_example/inception_client --server=localhost:9031 --image_file=/tmp/imagenet/cropped_panda.jpg:/tmp/imagenet/cropped_panda.jpg

Test Dynamic Serving:
#Copy new model to path in version 2
cp -r ~/matroid/scheduler/model_serving/<model-id-2>/00000001/ ~/matroid/scheduler/model_serving/<model-id>/00000002/ 
#Run the client script with same image
~/serving/bazel-bin/tensorflow_serving/inception_example/inception_client --server=localhost:9031 --image_file=/tmp/imagenet/cropped_panda.jpg:/tmp/imagenet/cropped_panda.jpg

You should get different results for the 2nd inference run as you are using a different model

### CODE REVIEW

**Comment by deepak-matroid** [comment]
Added missing grpc folder as linked submodule in serving

**Comment by deepak-matroid** [comment]
Issue #273 of matroid repo

File: tensorflow_serving/inception_example/inception_inference.proto [hunk 1]
```diff
@@ -37,6 +38,7 @@ message SingleRequest {
   // for video, only return frames that detector are confident
   float frame_detection_threshold = 4;
   bool return_detected_frames = 5;
+  string network_id = 6;
```
Comment (dong-matroid) [inline tensorflow_serving/inception_example/inception_inference.proto]: might remove this to keep the network_id in BatchRequest

**Comment by dong-matroid** [comment]
LGTM, thx @deepak-matroid

**Pull request merged**